test_vm/tests/commit_post_test.rs

```diff
@@ -0,0 +1,426 @@
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: *STORAGE_POWER_ACTOR_ADDR,
+                method: PowerMethod::SubmitPoRepForBulkVerify as u64,
+                ..Default::default()
+            },
+        ]),
+        ..Default::default()
+    }
+    .matches(v.take_invocations().last().unwrap());
+    let res = v
+        .apply_message(
+            *SYSTEM_ACTOR_ADDR,
+            *CRON_ACTOR_ADDR,
+            TokenAmount::zero(),
+            CronMethod::EpochTick as u64,
+            RawBytes::default(),
+        )
+        .unwrap();
+    assert_eq!(ExitCode::OK, res.code);
+    ExpectInvocation {
+        to: *CRON_ACTOR_ADDR,
+        method: CronMethod::EpochTick as u64,
+        subinvocs: Some(vec![
+            ExpectInvocation {
+                to: *STORAGE_POWER_ACTOR_ADDR,
+                method: PowerMethod::OnEpochTickEnd as u64,
+                subinvocs: Some(vec![
+                    ExpectInvocation {
+                        to: *REWARD_ACTOR_ADDR,
+                        method: RewardMethod::ThisEpochReward as u64,
+                        ..Default::default()
+                    },
+                    ExpectInvocation {
+                        to: id_addr,
+                        method: MinerMethod::ConfirmSectorProofsValid as u64,
+                        subinvocs: Some(vec![ExpectInvocation {
+                            to: *STORAGE_POWER_ACTOR_ADDR,
+                            method: PowerMethod::UpdatePledgeTotal as u64,
+                            ..Default::default()
+                        }]),
+                        ..Default::default()
+                    },
+                    ExpectInvocation {
+                        to: *REWARD_ACTOR_ADDR,
+                        method: RewardMethod::UpdateNetworkKPI as u64,
+                        ..Default::default()
+                    },
+                ]),
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: *STORAGE_MARKET_ACTOR_ADDR,
+                method: MarketMethod::CronTick as u64,
+                ..Default::default()
+            },
+        ]),
+        ..Default::default()
+    }
+    .matches(v.take_invocations().last().unwrap());
+    // pcd is released ip is added
+    let balances = v.get_miner_balance(id_addr);
+    assert!(balances.initial_pledge > TokenAmount::zero());
+    assert!(balances.pre_commit_deposit == TokenAmount::zero());
+
+    // power unproven so network stats are the same
+    let p_st = v.get_state::<PowerState>(*STORAGE_POWER_ACTOR_ADDR).unwrap();
+    assert_eq!(TokenAmount::zero(), p_st.total_bytes_committed);
+    assert!(p_st.total_pledge_collateral > TokenAmount::zero());
+    let (dline_info, p_idx, v) = advance_to_proving_deadline(v, id_addr, sector_number);
+
+    // submit post
+    let st = v.get_state::<MinerState>(id_addr).unwrap();
```

Comment:
You can consider changing the signature of the test to return `Result` to use the shorthand `?` instead of unwrapping.